### PR TITLE
[JB][SUPP-477] upgrade to Scala 2.13

### PR DESCRIPTION
**Pull request by barnesjake**

### CODE REVIEW

File: app/controllers/UpfrontPaymentController.scala [hunk 1]
```diff
@@ -88,11 +88,12 @@ class UpfrontPaymentController @Inject() (
       case j: Journey.BeforeAnsweredCanPayUpfront =>
         logErrorAndRouteToDefaultPage(j)
 
-      case j: Journey.AfterAnsweredCanPayUpfront if !j.canPayUpfront.userCanPayUpfront =>
-        logErrorAndRouteToDefaultPage(j)
-
-      case j: Journey.AfterAnsweredCanPayUpfront if j.canPayUpfront.userCanPayUpfront =>
-        finalStateCheck(j, displayUpfrontPaymentAmountPage(request.eligibilityCheckResult, Left(j)))
+      case j: Journey.AfterAnsweredCanPayUpfront =>
```
Comment (barnesjake) [inline app/controllers/UpfrontPaymentController.scala:91]: compiler made me do this, I prefer it as it was before, but it thinks there's possible a missed case, which I guess there technically could be but 🤷

File: app/models/audit/eligibility/EligibilityCheckAuditDetail.scala [hunk 2]
```diff
@@ -74,8 +74,9 @@ object EligibilityCheckAuditDetail {
               } else {
                 (currentFieldName, transformedJsArray)
               }
-            case value if currentFieldName === fieldName => (currentFieldName, transformation(value))
-            case value if currentFieldName =!= fieldName => (currentFieldName, value)
+            case value if currentFieldName === fieldName                         => (currentFieldName, transformation(value))
+            case value if currentFieldName =!= fieldName                         => (currentFieldName, value)
+            case value @ (JsNull | JsNumber(_) | JsString(_) | JsTrue | JsFalse) => (currentFieldName, value)
```
Comment (barnesjake) [inline app/models/audit/eligibility/EligibilityCheckAuditDetail.scala:79]: Compiler made me do it

File: app/models/forms/CanPayUpfrontForm.scala [hunk 3]
```diff
@@ -27,7 +27,7 @@ object CanPayUpfrontForm {
   def form(implicit language: Language): Form[CanPayUpfrontFormValue] = {
 
     val canPayUpfrontMapping: Mapping[CanPayUpfrontFormValue] = Forms.of(EnumFormatter.format(
-      enum                    = CanPayUpfrontFormValue,
+      `enum`                  = CanPayUpfrontFormValue,
```
Comment (barnesjake) [inline app/models/forms/CanPayUpfrontForm.scala:30]: Compiler made me do it, something to do with enum becoming a keyword in the future (scala 3 I think)

File: app/services/AuditService.scala [hunk 4]
```diff
@@ -124,8 +124,8 @@ class AuditService @Inject() (auditConnector: AuditConnector)(implicit ec: Execu
     val enrollmentReasons =
       if (eligibilityCheckResult.isEligible) None else Some(EnrollmentReasons.DidNotPassEligibilityCheck())
     val eligibilityReasons = {
-      val reasons = eligibilityCheckResult.eligibilityRules.getClass.getDeclaredFields.map(_.getName)
-      val values = eligibilityCheckResult.eligibilityRules.productIterator.to
+      val reasons: Array[String] = eligibilityCheckResult.eligibilityRules.getClass.getDeclaredFields.map(_.getName)
+      val values = eligibilityCheckResult.eligibilityRules.productIterator.iterator
```
Comment (barnesjake) [inline app/services/AuditService.scala:128]: Compiler made me do this, I don't think there's any breaking change to this functionality, as we have tests for it

File: app/services/BarsService.scala [hunk 5]
```diff
@@ -106,6 +107,7 @@ class BarsService @Inject() (barsConnector: BarsConnector)(implicit ec: Executio
       case accountNumberIsWellFormattedNo() => AccountNumberNotWellFormattedValidateResponse(response)
       case sortCodeIsPresentOnEiscdNo()     => SortCodeNotPresentOnEiscdValidateResponse(response)
       case sortCodeSupportsDirectDebitNo()  => SortCodeDoesNotSupportDirectDebitValidateResponse(response)
+      case ValidateResponse(_)              => Errors.throwServerErrorException("Fell into case ValidateResponse(_) in handleValidateErrorResponse. This should never happen, debug.")
```
Comment (barnesjake) [inline app/services/BarsService.scala:110]: Compiler made me do it, so I put in an error log

File: app/util/ImplicitConversions.scala [hunk 6]
```diff
@@ -18,7 +18,7 @@ package util
 
 import play.api.mvc.Result
 import scala.concurrent.Future
-import scala.language.implicitConversions
+//import scala.language.implicitConversions
```
Comment (barnesjake) [inline app/util/ImplicitConversions.scala:21]: I thought I deleted this, will update

File: app/models/audit/eligibility/EligibilityCheckAuditDetail.scala (hunk 2 above)
Comment (AndyHWChung) [reply app/models/audit/eligibility/EligibilityCheckAuditDetail.scala:79]: does something like this work to make it shorter?
```scala
case value if currentFieldName === fieldName => (currentFieldName, transformation(value))
case value: JsValue => (currentFieldname, value)
```

Comment (barnesjake) [reply app/models/audit/eligibility/EligibilityCheckAuditDetail.scala:79]: I'll try it, compiler is so damn strict now (which is a good thing 😂 )

Comment (barnesjake) [reply app/models/audit/eligibility/EligibilityCheckAuditDetail.scala]: Yeah it did, pushing now

**Review by AndyHWChung — APPROVED** [review]

**Pull request merged**